### PR TITLE
[Merged by Bors] - refactor: move morphisms in StructuredArrow to a lower universe

### PR DESCRIPTION
**Pull request by TwoFX**
This PR changes the definition
```lean
abbrev fromPUnit (X : C) : Discrete PUnit.{v + 1} ⥤ C :=
  (Functor.const _).obj X
```
to
```lean
abbrev fromPUnit (X : C) : Discrete PUnit.{w + 1} ⥤ C :=
  (Functor.const _).obj X
```
and then redefines
```lean
def StructuredArrow (S : D) (T : C ⥤ D) :=
  Comma (Functor.fromPUnit S) T
```
to
```lean
def StructuredArrow (S : D) (T : C ⥤ D) :=
  Comma (Functor.fromPUnit.{0} S) T
```

The effect of this is that given `{C : Type u₁} [Category.{v₁} C] {D : Type u₂} [Category.{v₂} D] (S : D) (T : C ⥤ D)`, the morphisms of `StructuredArrow S T` no longer live in `max v₁ v₂`, but in `v₁`, as they should. Thus, this PR is basically a better version of #6388.

My guess for why we used to have the larger universe is that back in the day, the universes for limits were much more restricted, so stating the results of `Limits/Comma.lean` was not possible if the morphisms of `StructuredArrow` live in `v₁`. Luckily, by now it is possible to state everything correctly, and this PR adjusts `Limits/Comma.lean` and `Limits/Over.lean` accordingly.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/PUnit.lean [hunk 1]
```diff
@@ -50,7 +50,7 @@ theorem punit_ext' (F G : C ⥤ Discrete PUnit) : F = G :=
 #align category_theory.functor.punit_ext' CategoryTheory.Functor.punit_ext'
 
 /-- The functor from `Discrete PUnit` sending everything to the given object. -/
-abbrev fromPUnit (X : C) : Discrete PUnit.{v + 1} ⥤ C :=
+abbrev fromPUnit (X : C) : Discrete PUnit.{1} ⥤ C :=
```
Comment (kbuzzard) [inline Mathlib/CategoryTheory/PUnit.lean:53]: How about 
```
abbrev fromUnit (X : C) : Discrete Unit ⥤ C :=
``` 
instead then? (plus some comment about why you don't want this to be polymorphic).

**Review by kbuzzard — APPROVED** [review]
In general I think that this is probably the right approach; I think it's good to have a bit more control over exactly which universes you're allowing limits over rather than just announcing `[HasLimits X]`.

File: Mathlib/CategoryTheory/PUnit.lean (hunk 1 above)
Comment (TwoFX) [reply Mathlib/CategoryTheory/PUnit.lean]: I ended up making the universe of `PUnit` fully explicit everywhere in the file and then fixing the universe when defining the category of (co)structured arrows. This made more sense to me than having a single result about `Unit` when the rest of the file is about `PUnit`.

**Comment by joelriou** [comment]
I also believe this is a nice improvement. Thanks!

bors merge

**Pull request closed without merging**